lib/Vaultaire/Collector/Common/Process.hs

```diff
@@ -203,6 +204,26 @@ collectSimple (SimplePoint addr ts payload) = do
             put (cS{ pointsBytesWritten = newLen}, eS)
             maybeRotatePointsFile
 
+-- | Wrapped Marquise.Client.queueExtended with logging.
+collectExtended :: MonadIO m => ExtendedPoint -> Collector o s m ()
+collectExtended (ExtendedPoint addr ts payload) = do
+    (cS@CommonState{..}, eS) <- get
+    res <- liftIO $ runMarquise $
+        queueExtended collectorSpoolFiles addr ts payload
+    case res of
+        Left e -> liftIO $ warningM "Process.collectExtended" $
+            "Marquise error when queuing extended point: " ++ show e
+        Right _ -> do
+            liftIO $ debugM "Process.collectExtended" $
+                concat ["Queued extended point "
+                       , show addr, ", "
+                       , show ts, ", "
+                       , show payload]
+            let payloadLen = fromIntegral $ BS.length payload
+            let newLen = pointsBytesWritten + 24 + payloadLen
```

Comment:
It took me a minute to work out why this was 24 and not 16 - explanatory comment here might be handy.